### PR TITLE
Refactor interfaces and event implementation into interfaces.py

### LINKED ISSUE #9: Move interfaces to interfaces.py
(But leave BWC imports in __init__.py.)

This follows the common standard and may also help with documentation.

### PR DESCRIPTION
**Pull request by jamadden**
Fixes #9 

Backwards compatibility aliases remain for the old names.

### CODE REVIEW

**Comment by jamadden** [comment]
Unless there are objections, I will probably merge this soon so I can move on with the other issues.

File: CHANGES.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@
 ==================
 
 - Add zope.lifecycleevent subscribers. You must include ``subscribers.zcml``
-  to use these. See https://github.com/zopefoundation/zc.intid/issues/5
+  to use these and have :mod:`zope.intid` installed. See :issue:`5`.
```
Comment (tseaver) [inline CHANGES.rst:9]: I don't know from the `:issue:` Sphinx role:  is it from an extension?  I don't see one [enabled](https://github.com/zopefoundation/zc.intid/blob/master/doc/conf.py#L32-L38) which might add it.

Comment (jamadden) [reply CHANGES.rst:9]: It's a [configured](https://github.com/zopefoundation/zc.intid/blob/master/doc/conf.py#L355) usage of the extlink extension. I've been using it in gevent to great effect (but then the gevent changelog contains *a lot* of links to issue/prs).

Comment (tseaver) [reply CHANGES.rst:9]: Good to know!  LGTM

Comment (jamadden) [reply CHANGES.rst:9]: Thanks!

**Pull request merged**